### DISCUSSION TITLE
Pyscript.net Documentation Enhancements PyCon2024

### DISCUSSION THREAD

**Discussion by nicksspirit** [pyscript/pyscript#2069] [Proposals]
# Update pyscript.net docs to encourage usage of pyscript-cli in user guide docs.
Currently there is no mention of the [pyscript-cli](https://github.com/pyscript/pyscript-cli). Since the pyscript-cli has been available for a year, it has made a lot of improvements to the developer workflow such as:

* Creating a new project
* Running a project with CORS headers already set.
* Various convenience commands such as wrapping an already existing python script etc.

I still believe we should keep the current documentation https://docs.pyscript.net/2024.5.2/beginning-pyscript/ for barebones setup but we should encourage the use of the cli to streamline the process for newcomers.

**Comment by nicksspirit** [comment]
@FabioRosado & @ntoll I am thinking we can streamline the getting started experience if we encourage the use of cli at least for the developers.

**Comment by ntoll** [comment]
This is good to see!

We have to be careful that we write for all the different "new users" that may encounter PyScript. Some of their needs will be mutually exclusive: e.g. a complete beginner will be lost and confused by a CLI, and likewise an experienced coder coming to PyScript for the first time won't want to be hand held.

**Comment by FabioRosado** [comment]
Just my two cents, but before we mention about the pyscript cli more, we should really really release the latest main to Pypi, once we do the release I am happy to advertise it more 😄

**Reply by nicksspirit** [reply]
Yes of course, we definitely should do that!

**Comment by nicksspirit** [comment]
@ntoll I agree with the discernment between beginner and advanced users which is why I am proposing an enhancement to the pyscript developer guide docs that provides both ways to get started building pyscript apps.

I think we can borrow a lot from fastapi with it's [CLI component](https://fastapi.tiangolo.com/fastapi-cli/). Obviously, fastapi-cli is different from the pyscript-cli nevertheless I think it can serve as a sort of muse or inspiration for how to write our developer guide docs.

**Reply by ntoll** [reply]
This is great, and I totally agree with the developer guide being the right place to put this sort of thing. :+1:
